### PR TITLE
Achapman/ch17993/reduce or eliminate the use of the controlled

### PR DESCRIPTION
**Pull request by nuevoalex**
A rather large PR that removes the Controller contract as well as the Controlled contract.

Rather than using a whitelist we're just looking for specific callers in most cases now. The registry component was moved to the Augur contract.

### CODE REVIEW

File: packages/augur-core/source/contracts/LegacyReputationToken.sol [hunk 1]
```diff
@@ -8,7 +8,7 @@ contract LegacyReputationToken is OldLegacyReputationToken {
     using ContractExists for address;
     event FundedAccount(address indexed _universe, address indexed _sender, uint256 _repBalance, uint256 _timestamp);
 
-    address private constant FOUNDATION_REP_ADDRESS = address(0x1985365e9f78359a9B6AD760e32412f4a445E862);
+    address private constant FOUNDATION_REP_ADDRESS = address(0xE94327D07Fc17907b4DB788E5aDf2ed424adDff6);
```
Comment (epheph) [inline packages/augur-core/source/contracts/LegacyReputationToken.sol]: Why change this?

Comment (nuevoalex) [reply packages/augur-core/source/contracts/LegacyReputationToken.sol]: Hm yea actually this isn't really needed. Will revert

File: packages/augur-core/source/contracts/Augur.sol [hunk 2]
```diff
@@ -60,10 +60,48 @@ contract Augur is Controlled, IAugur {
...
+            time = ITime(_address);
+        }
+        return true;
+    }
+
+    function lookup(bytes32 _key) public view returns (address) {
+        return registry[_key];
+    }
+
+    function finishDeployment() public returns (bool) {
```
Comment (epheph) [inline packages/augur-core/source/contracts/Augur.sol:101]: Anyone can call this, potentially breaking an upload process before it is finished?

Comment (nuevoalex) [reply packages/augur-core/source/contracts/Augur.sol:101]: fair point! Will restrict

**Review by epheph — APPROVED** [review]
🦉

**Review by justinbarry — APPROVED** [review]
LGTM

**Pull request merged**